pkg/networkservice/common/discover/match_selector.go

```diff
@@ -56,15 +36,19 @@ func matchEndpoint(clockTime clock.Clock, nsLabels map[string]string, ns *regist
 	// Iterate through the matches
 	for _, match := range ns.GetMatches() {
 		// All match source selector labels should be present in the requested labels map
-		if !isSubset(nsLabels, match.GetSourceSelector(), nsLabels) {
+		if !matchutils.IsSubset(nsLabels, match.GetSourceSelector(), nsLabels) {
 			continue
 		}
 		nseCandidates := make([]*registry.NetworkServiceEndpoint, 0)
 		// Check all Destinations in that match
 		for _, destination := range match.GetRoutes() {
 			// Each NSE should be matched against that destination
 			for _, nse := range validNetworkServiceEndpoints {
-				if isSubset(nse.GetNetworkServiceLabels()[ns.Name].Labels, destination.GetDestinationSelector(), nsLabels) {
```

Comment:
potential nil panic.

nse.GetNetworkServiceLabels()[ns.Name] can be nil. 

Fixed.